lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -515,6 +525,20 @@ def check_watched_item(self, job_state):
             # job is no longer viable - remove from watched jobs
             return None
 
+    def _handle_unschedulable_job(self, job, job_state):
+        # Handle unschedulable job that exceeded deadline
+        with open(job_state.error_file, 'a') as error_file:
```

Comment:
I think it would be good to guard against the condition where this file doesn't exist, because I've seen several instances (in other places in the runner) where writing to the error file fails.

Reply:
What do we need this file for ? Can we eliminate this ? This was used in older traditional HPC runners that would kill a job and then write something to a specific file, but kubernetes doesn't do that, and we can write to `job_state.fail_message`. I would assume that ends up on the job_stderr column of the job table.

Reply:
I was just following what is done to handle failures in other places eg https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/kubernetes.py#L521 . Can we just change this as part of moving to the pulsar runner? Not really worth the time sink now?